### PR TITLE
Resolve sidemenu comments

### PR DESCRIPTION
**Pull request by yksolanki9**
Fixes:
- Show option directly if only one option inside dropdown (Cards / Teams)
- Add divider between primary and secondary options
- Fix padding and alignment issues

localhost_8100_enterprise_team_reports_filters=%7B%22state%22_%5B%22APPROVER_PENDING%22%5D%7D(Samsung Galaxy S8+)

### CODE REVIEW

File: src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content-item/sidemenu-content-item.component.scss [hunk 1]
```diff
@@ -34,10 +34,15 @@
...
     &--active {
+      color: $pure-white;
+
       ion-icon {
         fill: $pure-white !important;
       }
-      color: $pure-white;
+
+      & > ion-grid {
+        padding-left: 9px;
```
Comment (Dimple16) [inline src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content-item/sidemenu-content-item.component.scss:44]: 10?

File: src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content.component.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ import { SidemenuItem } from 'src/app/core/models/sidemenu-item.model';
 export class SidemenuContentComponent implements OnInit {
   @Input() sideMenuList: Partial<SidemenuItem>[];
 
+  @Input() numPrimaryOptions: number;
```
Comment (Dimple16) [inline src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content.component.ts:17]: `primaryOptionsCount`?

**Review by Dimple16 — CHANGES_REQUESTED** [review]

File: src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content-item/sidemenu-content-item.component.scss (hunk 1 above)
Comment (yksolanki9) [reply src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content-item/sidemenu-content-item.component.scss:44]: The design uses a 3px red margin for active tabs. 4px/2px dont look good. So if I make this 10px, I'll have to make the others 13px

**Review by Dimple16 — APPROVED** [review]

File: src/app/shared/components/sidemenu/sidemenu.component.ts [hunk 3]
```diff
@@ -228,7 +231,35 @@ export class SidemenuComponent implements OnInit {
         disabled: !isConnected,
         dropdownOptions: teamOptions,
       },
-    ];
+    ].filter((sidemenuItem) => sidemenuItem.isVisible);
+
+    this.primaryOptionsCount = primaryOptions.length;
+
+    if (cardOptions.length === 1) {
+      primaryOptions.splice(
```
Comment (Julias0) [inline src/app/shared/components/sidemenu/sidemenu.component.ts:239]: Extract this and the below part into a pure function. @Dimple16 can help with this

File: src/app/shared/components/sidemenu/sidemenu.component.ts [hunk 4]
```diff
@@ -278,14 +309,13 @@ export class SidemenuComponent implements OnInit {
         route: ['/', 'enterprise', 'help'],
         disabled: !isConnected,
       },
-    ];
+    ].filter((sidemenuItem) => sidemenuItem.isVisible);
```
Comment (Julias0) [inline src/app/shared/components/sidemenu/sidemenu.component.ts:312]: why move this here?

Comment (yksolanki9) [reply src/app/shared/components/sidemenu/sidemenu.component.ts:312]: This was done to get count of primary options that would be visible and display the divider accordingly. For secondary options it wont matter if we filter in this method or after combining with primary.

File: src/app/shared/components/sidemenu/sidemenu.component.ts (hunk 3 above)
Comment (yksolanki9) [reply src/app/shared/components/sidemenu/sidemenu.component.ts]: Done

File: src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content.component.html [hunk 5]
```diff
@@ -1,8 +1,9 @@
-<div>
+<div class="sidemenu-content">
   <ion-list>
-    <ng-container *ngFor="let sidemenuItem of sideMenuList">
+    <ng-container *ngFor="let sidemenuItem of sideMenuList; index as i">
       <app-sidemenu-content-item
         [sidemenuItem]="sidemenuItem"
+        [showDivider]="i === primaryOptionsCount ? true : false"
```
Comment (Julias0) [inline src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content.component.html:6]: `[showDivider]="i === primaryOptionsCount"` should behave the same

**Review by Julias0 — APPROVED** [review]

File: src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content.component.html (hunk 5 above)
Comment (yksolanki9) [reply src/app/shared/components/sidemenu/sidemenu-content/sidemenu-content.component.html]: Done

**Pull request merged**